ray_data_eval/common/pipeline.py

```diff
@@ -0,0 +1,192 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class OperatorSpec:
+    name: str
+    operator_idx: int
+    num_tasks: int
+    duration: int
+    input_size: int
+    output_size: int
+    num_cpus: int
+
+
+@dataclass
+class TaskSpec:
+    id: str
+    operator_idx: int
+    duration: int
+    input_size: int
+    output_size: int
+    num_cpus: int
+
+
+def _get_tasks(operator_list: list[OperatorSpec]):
+    tasks = []
+    # Reversed so that downstream tasks are prioritized.
```

Comment:
This design looks a little brittle; I don't think the policies should rely on this ordering. It's fine to keep as is, but we should probably sort again in the policies.

Reply:
Added a todo: I think eventually the implementation of the policy should not rely on ordering.